src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -203,17 +203,47 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
     });
   }
 
+  // Solution for checking partial element visibility adapted from
+  // https://stackoverflow.com/questions/30943662/check-if-element-is-partially-in-viewport
+  /**
+   * Checks if a given element is in view.
+   *
+   * @params e element to perform check for
+   */
+  isInViewport(e: HTMLElement): boolean {
+    const rect: ClientRect = e.getBoundingClientRect();
+    const windowHeight: number = (window.innerHeight || document.documentElement.clientHeight);
+
+    return !(
+      Math.floor(100 - (((rect.top >= 0 ? 0 : rect.top) / +-rect.height) * 100)) < 1 ||
+      Math.floor(100 - ((rect.bottom - windowHeight) / rect.height) * 100) < 1
+    );
+  }
+
+  /**
+   * Scrolls to the question based on its given question id.
+   */
+  scrollToQuestion(): void {
+    const div: any = document.getElementById(this.moderatedQuestionId);
```

Comment:
Don't have to type as `any` here

Reply:
I initially went with type as `HTMLElement` but it resulted in this error message in instances where the element is null: `Type 'null' is not assignable to type 'HTMLElement'` - hence the choice of `any`. Are there alternatives to this?

Reply:
Hello @ccyccyccy! With reference to my comment above, would you have any advice or is it alright to keep the type `any`? >.< Thank You!

Reply:
Hi @tjtanjin why do you need to type it like that? Typescript should be able to infer the type: https://www.typescriptlang.org/docs/handbook/type-inference.html